### ISSUE TITLE
heading angle calculation own_dataset

### ISSUE THREAD

**Issue by vijaysamula** [facebookresearch/votenet#107]
Hi, I am Vijay. Votenet is simply superb and I am using it to detect people. I have point cloud, CSV file consisting of 
id,n_points,centroid_x,centroid_y,centroid_z,min_b_x,min_b_y,min_b_z,max_b_x,max_b_y,max_b_z,point_indices. The problem is I don't have heading angle. Would you guide me in finding the heading angle ?

**Comment by twzoechen** [comment]
@vijaysamula  have you solved the problem?

**Comment by vijaysamula** [comment]
No, I didn't use the heading angle. I just trained with a bounding box, votes.

**Comment by twzoechen** [comment]
@vijaysamula Can I have your contact information, such as email ?

**Comment by damonftl** [comment]
I'm wondering how this was accomplished. I'm training on my own data too, and heading angle is not required. I think it's actually negatively affecting the results. Do you just set that angle to zero for all training point clouds? Or did you remove it completely from the VoteNet network?

Currently I'm just setting it to zero for all my segmented objects, but I'm afraid it's causing problems with training. The best I can get right now so far is about 10% mAP. The objects I'm training in can face any direction, so I'd like to not use the heading angle at all.

**Comment by ljx398966986** [comment]
I labeled my data with an angle in 0-360°,but the final results is not pretty good.I don't know if my process on angle is wrong

**Comment by rXYZkit** [comment]
If you have the transformation matrix (or rotation matrix) of each object relative to the world coordinate system, you can get the Euler angle through the function in the scipy library. I think the Euler angle of the z axis * -1 is the heading_angle (assuming The world coordinate system is the upright depth coordinate system)

**Comment by EZreal-zhangxing** [comment]
i got same problem,i want to train on PROX dataset and i use the other algorithm(dbscan of open3d) calculate the target points and then calculate  oriented bounding box of target .with this worked i get the rotation matrix . rotation matrix can be transform to euler angle .but it's hard to understand the define of axis
